course/page/upload.py

```diff
@@ -46,11 +56,16 @@ class FileUploadForm(StyledForm):
     uploaded_file = forms.FileField(required=True,
             label=ugettext_lazy('Uploaded file'))
 
-    def __init__(self, maximum_megabytes, mime_types, *args, **kwargs):
+    def __init__(self, maximum_megabytes, mime_types,
+                 clean_uploaded_file_callback=None,
+                 *args, **kwargs):
         super(FileUploadForm, self).__init__(*args, **kwargs)
 
         self.max_file_size = maximum_megabytes * 1024**2
         self.mime_types = mime_types
+        self.clean_uploaded_file_callback = clean_uploaded_file_callback
+        if self.clean_uploaded_file_callback is not None:
+            assert callable(self.clean_uploaded_file_callback)
```

Comment:
This should be done by subclassing the form.

Reply:
I had that idea in the first place. Howeve, by using this callback, we do not need to do the subclassing each time we add a new type of question with different `mime_type` (I think the main difference is the clean method of the `uploaded_file`).

Reply:
I think you are right on this.